app/src/main/java/edu/uwstout/p2pchat/SettingNode.java

```diff
@@ -0,0 +1,387 @@
+    /**
+     * All subviews of theme color will be assigned here.
+     */
+    private LinkedList<TextView> childView2 = new LinkedList<>();
+
+    /**
+     * All subviews of nickname shall be assigned here. Only 0 will be assigned
+     * here.
+     */
+    private LinkedList<TextView> childView3 = new LinkedList<>();
+
+    /**
+     * All subviews of help will be assigned here.
+     */
+    private LinkedList<TextView> childView4 = new LinkedList<>();
+    //MAYBE
+    FragmentManager manager;
+    /**
+     * To save theme color, will be done with this variable.
+     */
+    private Save saveInfo;
+    /**
+     * Name of each cell of expandible list view will be in here.
+     */
+    private String[] parentNames = {
+            "My Device Info", "Theme Color: Light", "Set Device Nickname",
+            "Help"
+    };
+    private String[][] childNames;
+    private Activity mActivity;
+
+    /**
+     * The context will be stored in here.
+     */
+    private Context context;
+
+
+    /**
+     * Constructor for the setting nodes.
+     *
+     * @param context
+     *         context needed for making other views.
+     * @param childNames
+     *         needed to use
+     */
+    public SettingNode(final Context context, final String[][] childNames, final Activity activity,
+            final FragmentManager manager)
+    {
+        this.childNames = childNames;
+        this.context = context;
+        saveInfo = new Save(this.context);
+        mActivity = activity;
+        this.manager = manager;
+
+        for (int general = 0; general < 4; general++)
+        {
+            //Assigns textviews to parent views.
+            parentView[general] = createTextView(parentNames[general], true);
+
+            //Assigns childviews textviews.
+            for (int specific = 0; specific < childNames[general].length; specific++)
+            {
+                TextView view = createTextView(childNames[general][specific], false);
+
+                //Determines what view gets what.
+                switch (general)
+                {
+                    case 0:
+                        childView1.add(view);
+                        break;
+                    case 1:
+                        childView2.add(view);
+                        break;
+                    case 2:
+                        childView3.add(view);
+                        break;
+                    case 3:
+                        childView4.add(view);
+                        break;
+                }
+            }
+        }
+
+        //Sets actionlisteners for theme color and nickname parents.
+        setNickNameActionListener();
+        setThemeColorListener();
+
+        //Changes both parent
+        changeTheme(saveInfo.getThemeColorThatShallBeSelected());
+
+
+    }
+
+    /**
+     * Gets the textview of the child.
+     * Indexs at childView[x] shall be
+     * parent = x - 1
+     */
+    public TextView getView(int parent, int index)
```

Comment:
with array lists, this would be one line of code: (assume the arrayList of childViews is called arr)
arr.get(parent).get(index);